### ISSUE TITLE
bad argument #3 to '?' (torch.*Tensor expected, got nil)

### ISSUE THREAD

**Issue by Ashjawik** [fstrub95/Autoencoders_cf#10]
Hey I am running into following Issue:

th main.lua  -file ../data/ml-10M100K/movieLens-10M.t7 -conf ../conf/conf.movieLens.10M.U.lua  -save network.t7 -type U -meta 1 -gpu 1
Options: 	
 - gpu  	 : 1	
 - file  	 : ../data/ml-10M100K/movieLens-10M.t7	
 - seed  	 : 0	
 - save  	 : network.t7	
 - conf  	 : ../conf/conf.movieLens.10M.U.lua	
 - type  	 : U	
 - meta  	 : 1	
Load training configuration...	
Load data...	
Loading data from disk...	
	
Users loaded      : 69878	
Items loaded      : 10677	
	
No Train ratings  : 8999073	
No Test  ratings  : 1000981	
Training ratio    : 0.89990244052682	
Training density  : 0.012061687313886	
	
Loading cunn...	
Loading data to GPU...	
Unbias the data...	
Data was successfully preloaded...	
Training network	
Start training : layer1-1	
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> output]
  (1): cfn.AppenderSparseOut
  (2): nnsparse.Densify
  (3): nn.Linear(10677 -> 700)
  (4): nn.Tanh
  (5): cfn.AppenderOut
  (6): nn.Linear(700 -> 10677)
  (7): nn.Tanh
}
 [................... 1/50 .....................] ETA: 0ms | Step: 0ms          
/home/xyc/torch-cl/install/bin/luajit: bad argument #3 to '?' (torch.*Tensor expected, got nil)
stack traceback:
	[C]: at 0x7f25703c5210
	[C]: in function '__newindex'
	misc/AutoEncoderTrainer.lua:71: in function 'Train'
	misc/AutoEncoderTrainer.lua:266: in function 'Execute'
	misc/TrainNetwork.lua:150: in function 'TrainNetwork'
	main.lua:69: in main chunk
	[C]: in function 'dofile'
	...o/torch-cl/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00405e90

Any idea or solution?
Thanks and Regards

**Comment by polajnta** [comment]
try adding a return to line 64 of ClassicLoader.lua

`return denseTensor:indexCopy(1, index:long(), data)`

and re-run data.lua

**Comment by Ashjawik** [comment]
Hey, thanks for looking into that. Your solution doesn't work for that issue. Same error appears when trying to train the Uencoder

**Comment by fstrub95** [comment]
Hi!
How did you preprocess your data (data.lua)? It looks like the meta data are missing
Florian

**Comment by Ashjawik** [comment]
Yes your are right. Meta data were missing. Thanks for your help

**Comment by fstrub95** [comment]
np!
